frontend/src/components/Dashboard/DashboardResultsTable.tsx

```diff
@@ -80,46 +98,66 @@ const DashboardResultsTable = (props: any) => {
-                            <td
-                                className={`text-center py-4 ${exercise.status === "not submitted" ? "cursor-default" : "cursor-pointer"} `}
-                                role={exercise.status !== "not submitted" ? "button" : ""}
-                                key={j}
-                                onClick={exercise.status !== "not submitted" ? (() => {
-                                    setModalContent(modal(exercise, student))
-                                    setShowModal(true)
-                                }) : (
-                                    undefined
-                                )}
-                            >
-                                <StatusBadge status={exercise.status} />
-                            </td>
-                        ))
-                        }
-                    </tr >
-                ))
-                }
+                        {student.exercises.map((exercise: any, j: number) => {
+                            const exerciseGrade = sortedExercises ? sortedExercises[j].grade : 0;
+                            let sumOfCoefficient = 0;
+                            let dividendTestGrade = 0;
+
+                            sessionGrade += exerciseGrade;
+
+                            console.log(student.exercises[j].exercise_title, student.exercises[j].testResults)
+                            student.exercises[j].testResults.forEach((testResult: any) => {
+                                sumOfCoefficient += testResult.exercise_test.coefficient || 0;
+                                if (testResult?.status === "success") {
+                                    dividendTestGrade += exerciseGrade * (testResult.exercise_test.coefficient || 0);
+                                }
+                            });
+
+                            let finalExerciseGrade = 0;
+                            if (sumOfCoefficient !== 0) {
+                                finalExerciseGrade = dividendTestGrade / sumOfCoefficient;
+                            }
+
+                            finalSessionGrade += finalExerciseGrade;
+
+                            return (
+                                <td
+                                    className={`text-center py-4 ${exercise.status === "not submitted" ? "cursor-default" : "cursor-pointer"} `}
+                                    role={exercise.status !== "not submitted" ? "button" : ""}
+                                    key={j}
+                                    onClick={exercise.status !== "not submitted" ? () => {
+                                        setModalContent(modal(exercise, student));
+                                        setShowModal(true);
+                                    } : undefined}>
+                                    <StatusBadge status={exercise.status} />
+                                    <GradeBadge grade={finalExerciseGrade} total={exerciseGrade} />
```

Comment:
A potential improvement can be hiding the GradeBadge when the exercice's status is "not submitted", as the grade is obviously 0, and bound to change.

Reply:
Done